### DISCUSSION TITLE
Slower than expected queries with Rails 7

### DISCUSSION THREAD

**Discussion by jshirley** [planetscale/discussion#394] [Q&A] [answered]
Hi folks,

First off, Planetscale is rad :)

I'm doing some testing with Planetscale, and have been pretty impressed so far with the overall offering. I'm testing against a Rails 7 app with a pretty well-developed set of fixtures and demo data, and can easily test against local SQLite, a pretty bog standard postgres instance (the production setup), and Planetscale.

Populating a small demo data set in SQLite benchmarks at 11.3ms and pointing the same code at Planetscale (only change is in the `database.yml`) it takes 11,130ms. 11 seconds vs 11ms is so surprising it feels that I've done something wrong or have something very incorrectly configured.

Running a very simple count query over 30 records takes between 45-50ms in the console, here's a pretty average example:
```
irb(main):007:0> Thing.count
  Thing Count (46.0ms)  SELECT COUNT(*) FROM `things` WHERE `things`.`archived` = FALSE
=> 29 
irb(main):009:0> Organization.count
  Organization Count (65.6ms)  SELECT COUNT(*) FROM `organizations`
=> 3 
```

This is why I think I have something misconfigured on my side, because selecting count(*) over 3 records should be pretty quick unless there is a lot of overhead.

The planetscale db is in us-west-1, and I'm on the west coast. I also have tried using the local proxy through `pscale connect` and pointing the local Rails app at that and queries are still slow (simple AR queries 40-50ms) but it isn't showing re-establishing connections or establishing new connections (at least not what's available with `--debug`) 

We haven't yet launched so I have quite a bit of freedom in both what I'm testing with, the configuration and setup. Hoping I can figure out more on the performance side here.

Thank's all!

Edit: I should have included this originally, but using `mtr` indicates that _maybe_ it's the hops to `us-west.connect.psdb.cloud`. The average latency there matches the 40-50ms, but my average against other `us-west-1` hosts is consistently at 20ms.

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @jshirley,

It's likely all network time from your laptop to the database.

Could you try running the same queries from the web console? It will show the query time including only the network hop within our network.

Our API is a Rails app as well. For local development we use a local MySQL. Then for production, you'll want your database in the same region as wherever your application is deployed.
